lib/api/model/events.dart

```diff
@@ -708,6 +709,40 @@ class SubscriptionPeerRemoveEvent extends SubscriptionEvent {
   Map<String, dynamic> toJson() => _$SubscriptionPeerRemoveEventToJson(this);
 }
 
+/// A Zulip event of type `user_status`: https://zulip.com/api/get-events#user_status
+@JsonSerializable(fieldRename: FieldRename.snake)
+class UserStatusEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'user_status';
+
+  final int userId;
+
+  @JsonKey(readValue: _readChange, includeToJson: false)
+  final UserStatusChange change;
+
+  static Object? _readChange(Map<dynamic, dynamic> json, String key) {
+    assert(json is Map<String, dynamic>); // value came through `fromJson` with this type
+    json['status_text'] as String?;
+    json['emoji_name'] as String?;
+    json['emoji_code'] as String?;
+    json['reaction_type'] as String?;
+    return json;
```

Comment:
Are these checks needed?

It looks like the next thing that will happen with the return value is it'll get passed to UserStatusChange.fromJson, and it will make the same casts. So I think everything would behave exactly the same if this just returned `json` with no checks.

That would also leave it to UserStatusChange to describe exactly what its expectations on the incoming JSON are, which seems appropriate.